### PR TITLE
Add tabbing for files actions menu

### LINKED ISSUE #36989: [BITV] 9.2.4.3/3.3 - When the menu "Actions" is open, it is still possible to navigate through the elements on the page behind the menu, causing visually hidden elements to receive keyboard focus in some cases. It is best to close the menu when leaving it. (3)
- [ ] Fix
- [ ] Backport

a0992f1dcb597171115501ed414d9bd3

<details>
<summary>Details</summary>

https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-89806fe784-v3-n3

</details>

- [ ] 9.2.4.3/8.2 - https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-89806fe784-v8-n2
- [ ] 9.2.4.3/10.3 - https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-89806fe784-v10-n3

### PR DESCRIPTION
**Pull request by nfebe**
Resolves : https://github.com/nextcloud/server/issues/36989

## Before 

[before.webm](https://github.com/nextcloud/server/assets/14317775/b91cf1ee-9bd1-4aa2-ae3a-4bdda9514f53)

## After

 [after.webm](https://github.com/nextcloud/server/assets/14317775/f49f29fe-891f-4242-8733-f1f9597f9638)

### CODE REVIEW

**Comment by Fenn-CS** [comment]
I should have put this in draft first, sorry about that folks but this is not ready for review as the implementation need to be tested with at least one modal for correctness and in this case that is https://github.com/nextcloud/server/issues/36989. (See todo in PR description)

**Comment by Pytal** [comment]
I believe this is functionally identical to https://github.com/focus-trap/focus-trap?

If so then we can use the lib instead i.e. https://github.com/nextcloud/server/pull/37780

**Comment by Fenn-CS** [comment]
Thanks @Pytal! Did not know of this. Even though it was tricky to write this, that looks solid!

One question, `focus-trap` library deactivates clicks outside the focus container. The way most modals and elements that need focus trap is when you click outside the container, the modals would close, would the new behavior be okay?

File: apps/files/js/fileactions.js [hunk 1]
```diff
@@ -373,18 +373,24 @@
...
+			const menuFocusTrap = OCA.createFocusTrap(menu.$el.get(0), {
+				allowOutsideClick: true,
+				fallbackFocus: menu.$el,
+			});
 
 			menu.$el.on('afterHide', function() {
 				context.$file.removeClass('mouseOver');
 				$trigger.removeClass('open');
 				$trigger.attr('aria-expanded', 'false');
+				menuFocusTrap.deactivate()
```
Comment (Fenn-CS) [inline apps/files/js/fileactions.js:387]: This only works when the user clicks on the same button that triggered files menu. If the user clicks else-where on the document which closes the menu, proper focus is not restored.

PS : @susnux @Pytal do you know where "background click closes menu" is handled?

**Comment by skjnldsv** [comment]
Kinda the same, files to vue already fixes it :see_no_evil:

**Pull request closed without merging**